app/views/annotations/create.js.erb

```diff
@@ -23,7 +26,14 @@ resultComponent.addAnnotation({
     start: <%= [@annotation.y1, @annotation.y2].min || 'undefined' %>,
     end: <%= [@annotation.y1, @annotation.y2].max || 'undefined' %>
   }
-});
+},
+<%= @annotation.annotation_text.annotation_category.flexible_criterion_id || 'undefined' %>,
```

Comment:
So this is interesting; basically, we want to avoid unnecessary queries, especially with the additional parameters you've added.

So you want to refactor this code so that the additional arguments are only passed if there annotation text has a *non-nil and non-zero deduction*. In the other cases, none of the additional arguments are required, which will save some computational time.